### PR TITLE
:tada: add the ability to update gist only if changed

### PR DESCRIPTION
**Pull request by MishaKav**
:tada: add the ability to update gist only if changed

I really like your action and I'm using it a lot in my projects.
I decided to contribute to this really cool feature.
I tried to keep your style and convention.
I added documentation and test my branch with all possible options, you can view it [here](https://github.com/MishaKav/api-testing-example/actions/workflows/dynamic-badges-action.yml) 

**`updateIfChanged` Feature:**
- add a new property `updateIfChanged: true`
- the default is `false` which gives backward compatibility
- documentation updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   },
   "keywords": [],
   "author": "",
-  "license": "ISC",
+  "license": "MIT",
```
Comment (MishaKav) [inline package.json:15]: update license according to https://github.com/Schneegans/dynamic-badges-action/blob/master/LICENSE

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "dynamic-badges-action",
-  "version": "1.0.0",
+  "version": "1.5.0",
```
Comment (MishaKav) [inline package.json:3]: add minor bump to next version

**Comment by Schneegans** [comment]
Hi there! Thank you very much! I'll test this ASAP.

**Comment by Schneegans** [comment]
It looks pretty good, I'll merge it to a separate branch in order to be able to test it thoroughly. If everything works as expected, I'll merge this to `master`, close #16 and push a new tag! Thank you so much :smiley:

**Comment by Schneegans** [comment]
It seems to work flawlessly! I'll do some minor refactoring and add some comments here and there.

I am not sure why anybody would want to update the gist if nothing changed, so I think we should make this actually the default behavior. I think I'll rename the parameter to `forceUpdate`. It will default to `false` and if set to `true`, the gist will be overridden in any case. Or was there a specific reason for you to disable this feature per default?

File: package.json (hunk 1 above)
Comment (Schneegans) [reply package.json:15]: Thanks for this fix!

File: package.json (hunk 2 above)
Comment (Schneegans) [reply package.json:3]: Also thanks for this fix :wink:

**Comment by MishaKav** [comment]
I'm not enabling it by default, only for the reason of "backward compatibility", didn't want to change something that thousands of users are used to.
But in general, I totally agree with you, that it should be like the default behavior.

**Pull request merged**